subprojects/tooling-api/src/main/java/org/gradle/tooling/model/idea/IdeaContentRoot.java

```diff
@@ -58,6 +58,40 @@
     @Incubating
     DomainObjectSet<? extends IdeaSourceDirectory> getGeneratedTestDirectories();
 
+    /**
+     * The set of resource directories.
+     * NOTE: The resources directory is only available for Java projects, otherwise it is empty set.
+     *
+     * @since 4.7
+     */
+    @Incubating
+    DomainObjectSet<? extends IdeaSourceDirectory> getResourceDirectories();
```

Comment:
For all new methods declare that they throw `UnsupportedOperationException` when invoked on an older Gradle distribution.